workspace-manager-clienttests/src/main/java/scripts/testscripts/PrivateControlledGcsBucketLifecycle.java

```diff
@@ -0,0 +1,273 @@
+  }
+
+  private CreatedControlledGcsBucket attemptCreatePrivateBucket(
+      ControlledGcpResourceApi resourceApi) throws Exception {
+    String jobId = UUID.randomUUID().toString();
+    var creationParameters =
+        new GcsBucketCreationParameters()
+            .name(bucketName)
+            .location(BUCKET_LOCATION)
+            .defaultStorageClass(GcsBucketDefaultStorageClass.STANDARD)
+            .lifecycle(new GcsBucketLifecycle().rules(Collections.emptyList()));
+
+
+    var privateUser = new PrivateResourceIamRoles();
+    privateUser.add(ControlledResourceIamRole.WRITER);
+    var commonParameters =
+        new ControlledResourceCommonFields()
+            .name(resourceName)
+            .cloningInstructions(CloningInstructionsEnum.NOTHING)
+            .accessScope(AccessScopeEnum.PRIVATE_ACCESS)
+            .privateResourceUser(
+                new PrivateResourceUser()
+                    .userName(privateResourceUser.userEmail)
+                    .privateResourceIamRoles(privateUser))
+            .managedBy(ControlledResourceCommonFields.ManagedByEnum.USER)
+            .jobControl(new JobControl().id(jobId));
+
+    var body =
+        new CreateControlledGcsBucketRequestBody()
+            .gcsBucket(creationParameters)
+            .common(commonParameters);
+
+    logger.info(
+        "Attempting to create bucket {} jobId {} workspace {}",
+        bucketName,
+        jobId,
+        getWorkspaceId());
+    CreatedControlledGcsBucket bucket = resourceApi.createBucket(body, getWorkspaceId());
+    while (ClientTestUtils.jobIsRunning(bucket.getJobReport())) {
+      TimeUnit.SECONDS.sleep(CREATE_BUCKET_POLL_SECONDS);
+      bucket = resourceApi.getCreateBucketResult(getWorkspaceId(), jobId);
+    }
+    logger.info("Create bucket status is {}", bucket.getJobReport().getStatus().toString());
+    return bucket;
+  }
+
+  private DeleteControlledGcsBucketResult deleteBucketAttempt(
+      ControlledGcpResourceApi resourceApi, UUID resourceId) throws Exception {
+    String deleteJobId = UUID.randomUUID().toString();
+    var deleteRequest =
+        new DeleteControlledGcsBucketRequest().jobControl(new JobControl().id(deleteJobId));
+    logger.info("Deleting bucket resource id {} jobId {}", resourceId, deleteJobId);
+    DeleteControlledGcsBucketResult result =
+        resourceApi.deleteBucket(deleteRequest, getWorkspaceId(), resourceId);
+    while (ClientTestUtils.jobIsRunning(result.getJobReport())) {
+      TimeUnit.SECONDS.sleep(DELETE_BUCKET_POLL_SECONDS);
+      result = resourceApi.getDeleteBucketResult(getWorkspaceId(), deleteJobId);
+    }
+    return result;
+  }
+
+  @Override
+  protected void doCleanup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+      throws ApiException {
+    super.doCleanup(testUsers, workspaceApi);
+    if (bucketName != null) {
+      logger.warn("Test failed to cleanup bucket " + bucketName);
```

Comment:
why doesn't this rely on janitor? When are we going to do anything about this warning?

Reply:
The client tests never directly create cloud resources, they do so through WSM (which should have Janitor configured in the environments we test against). Removed the warning, since it's not actionable